# EDA Project: Used Car Market Analysis 
**Section: Python Data Wrangling & Exploratory Data Analysis**

---

## 1. Project Background & Objectives
This project aims to analyze a large-scale used car transaction dataset to understand key factors influencing vehicle market prices, brand distributions, and to identify data anomalies prior to modeling or advanced analytics.

### Key Business Questions:
1. What is the size and initial quality of the dataset (missing values & data types)?
2. Which car manufacturers dominate the used car market supply?
3. How does the transmission type correlate with the average selling price?

## 2. Environment Setup & Data Ingestion
In this section, we import the necessary Python libraries for data manipulation (`pandas`, `numpy`) and load the raw dataset into a Pandas DataFrame.

In [ ]:
import pandas as pd
import numpy as np

# Format floating-point numbers to 2 decimal places for better readability
pd.set_option('display.float_format', lambda x: '%.2f' % x)

# Load the dataset
df = pd.read_csv('data/vehicles.csv')

print("Dataset successfully loaded!")

Dataset successfully loaded!


## 3. Data Inspection & Quality Audit
The goal here is to inspect the dataset's dimensions, preview sample records, check data types, and identify missing values across all columns.

In [ ]:
# Check dataset dimensions (rows, columns)
print(f"Total Rows: {df.shape[0]}")
print(f"Total Columns: {df.shape[1]}")

# Preview the first 5 rows
df.head()

Total Rows: 426880
Total Columns: 26


,id,url,region,region_url,price,year,manufacturer,model,condition,cylinders,...,size,type,paint_color,image_url,description,county,state,lat,long,posting_date
0,7222695916,https://prescott.craigslist.org/cto/d/prescott...,prescott,https://prescott.craigslist.org,6000,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,az,NaN,NaN,NaN
1,7218891961,https://fayar.craigslist.org/ctd/d/bentonville...,fayetteville,https://fayar.craigslist.org,11900,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,ar,NaN,NaN,NaN
2,7221797935,https://keys.craigslist.org/cto/d/summerland-k...,florida keys,https://keys.craigslist.org,21000,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,fl,NaN,NaN,NaN
3,7222270760,https://worcester.craigslist.org/cto/d/west-br...,worcester / central MA,https://worcester.craigslist.org,1500,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,ma,NaN,NaN,NaN
4,7210384030,https://greensboro.craigslist.org/cto/d/trinit...,greensboro,https://greensboro.craigslist.org,4900,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,nc,NaN,NaN,NaN


In [ ]:
# Check data types and non-null counts
df.info()

# Count total missing values per column
df.isnull().sum()

<class 'pandas.DataFrame'>
RangeIndex: 426880 entries, 0 to 426879
Data columns (total 26 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   id            426880 non-null  int64  
 1   url           426880 non-null  str    
 2   region        426880 non-null  str    
 3   region_url    426880 non-null  str    
 4   price         426880 non-null  int64  
 5   year          425675 non-null  float64
 6   manufacturer  409234 non-null  str    
 7   model         421603 non-null  str    
 8   condition     252776 non-null  str    
 9   cylinders     249202 non-null  str    
 10  fuel          423867 non-null  str    
 11  odometer      422480 non-null  float64
 12  title_status  418638 non-null  str    
 13  transmission  424324 non-null  str    
 14  VIN           265838 non-null  str    
 15  drive         296313 non-null  str    
 16  size          120519 non-null  str    
 17  type          334022 non-null  str    
 18  paint_color   2

id                   0
url                  0
region               0
region_url           0
price                0
year              1205
manufacturer     17646
model             5277
condition       174104
cylinders       177678
fuel              3013
odometer          4400
title_status      8242
transmission      2556
VIN             161042
drive           130567
size            306361
type             92858
paint_color     130203
image_url           68
description         70
county          426880
state                0
lat               6549
long              6549
posting_date        68
dtype: int64

> 📝 **Data Audit Summary:**
> - **Dataset Size:** 426,880 rows and 26 columns.
> - **Missing Values:** `county` contains the highest number of missing records, followed by `size` and `cylinders`.
> - **Data Types:** Core numerical columns such as `price` and `odometer` are correctly stored as numerical types (`int64` / `float64`), eliminating the need for string parsing on these key features.